src/gui/EntryTree.cpp

```diff
@@ -626,12 +628,22 @@ void EntryTree::removeEntryRecurse(QTreeWidgetItem* item) {
 
 void LoadPackFileWorker::run(EntryTree* tree, const PackFile& packFile) {
 	int progress = 0;
+    // Neither the tree nor the model uses it's signals as items are being put in.
+    // Firing signals slows the item creation process by a whole 50%! So we disable it.
+    // During the creation process and turn it back on when it's done.
+    tree->blockSignals(true);
+    tree->model()->blockSignals(true);
+
 	for (const auto& [directory, entries] : packFile.getBakedEntries()) {
 		emit progressUpdated(++progress);
 		for (const auto& entry : entries) {
 			tree->addNestedEntryComponents(QString(directory.c_str()) + '/' + entry.getFilename().c_str());
 		}
 	}
+
+    tree->blockSignals(false);
+    tree->model()->blockSignals(false);
+
 	tree->sortItems(0, Qt::AscendingOrder);
```

Comment:
Should signals be disabled until after sorting is complete?

Reply:
It doesn't make any difference performance wise.